cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs

```diff
@@ -873,6 +878,12 @@ exResolve
       toConstraint (ExStanzaConstraint scope stanzas) =
         toLpc $ PackageConstraint scope (PackagePropertyStanzas stanzas)
 
+      parseConstraint :: String -> LabeledPackageConstraint
+      parseConstraint str =
+        case explicitEitherParsec parsec str of
+          Left err -> error ("Bad user constraint: " ++ err)
+          Right user_constraint -> toLpc (userToPackageConstraint user_constraint)
```

Comment:
This seems like an added benefit to me: the ability to easily add user constraints to the solver testsuite in the way that they are typically written, which doubles as a validation of the syntax parser. (unlike e.g. specifying (installed) packages and their dependencies for which the DSL is amazing to write concise tests, I don't think we'd win much by using a different syntax for user constraints which are already their own "canonical" concise way of being specified)

Reply:
I don't think that we should test `UserConstraint`s or parsing in this test suite, since that functionality is part of the cabal-install package, and these unit tests are for the dependency solver. These tests should eventually be moved to the cabal-install-solver package as a followup to #7358. I also think that it is confusing to have two ways to specify the same constraints.

Reply:
I see how that is reasonable. I'm going to drop the UserConstraint syntax testsuite support and change the tests to use PackageConstraints directly.